### PR TITLE
Make orders editable and deletable.

### PR DESCRIPTION
**Pull request by zestyping**

### CODE REVIEW

**Comment by dancunningham** [comment]
@zestyping can you make a corresponding pull request on the client app for this?

**Comment by dancunningham** [comment]
Found it! https://github.com/projectbuendia/client/commit/fb2949bb8c6d3764b46269f56cffb97f25c8f104

It's part of the scratch/chad-bokoro branch. @llvasconcellos is going to try and merge it along with some of the other things in that branch.

**Comment by llvasconcellos** [comment]
I merged dev into this and applied changes to make orders editable and "deletable" according to the most recent context.
It's working for edit and delete observations.
@capnfabs Can you have a look at this PR and see if I did not alter anything you worked on the past PRs? There were a lot of your commits to OrderResource.java. I did my best to keep only to change what we needed to make this work.

File: openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/OrderResource.java [hunk 1]
```diff
@@ -341,6 +409,30 @@ private Order reviseOrder(Order order, SimpleObject edits) {
                     }
                     break;
 
+                case "start_millis":
```
Comment (llvasconcellos) [inline openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/OrderResource.java]: @capnfabs On Ping's initial commit for this functionality he removed the case "stop" from the switch. I kept it. Do we need it?

**Comment by capnfabs** [comment]
Hmmm. This is going to need a little more thought if we want it to work with our current sync model.

**Comment by capnfabs** [comment]
I think there's some good reasons behind why Orders in OpenMRS were modelled like this. Editing orders should only be supported if the order hasn't been executed before, and is fairly recent. Otherwise, it's important to display the old order as well as the new order, for the cases when the user _isn't_ just editing a typo, but is changing a dosage or something.

So, here's my proposal:

- Only allow edits and deletions if less than a day has passed and there's no executions of the order. When an edit occurs, the old order is voided and a new order is created in its stead.
- Beyond that, allow revisions. When a revision occurs, we keep both the old order and the new order. Both of these are displayed on the client.

File: openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/OrderResource.java (hunk 1 above)
Comment (capnfabs) [inline openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/OrderResource.java]: It looks like Ping changed the field name from "stop" to "stop_millis" here, so it's ok to remove the "stop".

**Comment by dancunningham** [comment]
Closing as this is superseded by #48

**Pull request closed without merging**